setup.py

```diff
@@ -1,50 +1,22 @@
-#!/usr/bin/env python
-import io
 import re
 
-from setuptools import find_packages, setup
+from setuptools import setup
 
-with io.open('./rubicon/objc/__init__.py', encoding='utf8') as version_file:
+with open('./rubicon/objc/__init__.py', encoding='utf8') as version_file:
     version_match = re.search(r"^__version__ = ['\"]([^'\"]*)['\"]", version_file.read(), re.M)
     if version_match:
         version = version_match.group(1)
     else:
         raise RuntimeError("Unable to find version string.")
 
 
-with io.open('README.rst', encoding='utf8') as readme:
-    long_description = readme.read()
-
-
 setup(
-    name='rubicon-objc',
     version=version,
```

Comment:
setup.cfg allows for direct attribute access; did you try:
```
version = attr: rubicon.objc.__version__
```
This should let us move setup.py back to being a bare stub.

Reply:
Right, the `attr: ...` directive is very useful - when it works. The issue is that `attr: ...` is implemented as "using the current Python interpreter, import the module and read the requested attribute at runtime". This fails for everything except simple pure Python modules with no dependencies (see pypa/setuptools#1724 and pypa/setuptools#1960). Although Rubicon is pure-Python and formally doesn't have dependencies (i. e. no `install_requires`), we do have import-time dependencies in the form of native libraries that are only available on macOS (libobjc, Core Foundation, Foundation - see #150). If we were to use `version = attr: rubicon.objc.__version__`, it would be impossible to build Rubicon from source on any system other than macOS. This would be an issue for example with our current GitHub Actions workflows, which assume that all steps other than running the main test suite can be performed on Linux.

pypa/setuptools#1960 suggests changing the `attr: ...` directive to read attributes statically using the `ast` module instead (where possible), which would allow us to use `attr: ...` safely. There hasn't been any response to that issue yet though, and no one has submitted a PR for it yet.

Reply:
/me grumbles. I was afraid that might be a problem. 

An alternative approach would be to double-down on using setup.cfg, and use `importlib.metadata` to extract the version from the wheel metadata at runtime. That would let us use a single declaration in the setup.cfg.

However, that doesn't need to be a blocker on merging this; the config you have here is totally sufficient IMHO.